### DISCUSSION TITLE
Adding new custom blocks in the new form builder

### DISCUSSION THREAD

**Discussion by emanuellopess** [impress-org/givewp#7470] [Developer Q&A]
Hello, we are building an extension for GiveWP new form builder and we are trying to understand how our blocks can be compatible and appear in the "build" category like a custom field.

**Comment by kjohnson** [comment]
Hey @emanuellopess, what kind of block are you looking to add?

Would you describe it more as a display **element** or a **field** for collecting input?

**Reply by emanuellopess** [reply]
@kjohnson Hello we are building a field for collecting input, a checkbox more specifically

**Reply by emanuellopess** [reply]
@kjohnson Is there much difference between both implementations? We tried to follow some code we found in GiveWP plugin but without much success.

We used the hook 'givewp_donation_form_block_render' and made a guttenberg block with the features we need.

**Reply by kjohnson** [reply]
With Form Field Manager installed you can add a checkbox to the form and then hook into the form submissions with a few different hooks:
- `givewp_donation_form_processing_start`
- `givewp_donate_controller_donation_created`

Alternatively, if you are creating an opt-in checkbox, like for a newsletter integration for example, I can add you to a private repository to use as an example for building a "form extension".

**Reply by emanuellopess** [reply]
@kjohnson Thanks for the help, unfortunately we need the info of the checkbox for our plugin to process a back-end hook, we are building a plugin to help with brazilian taxes and receipt.

May you add me in this private repository?

**Reply by kjohnson** [reply]
@emanuellopess can you send me an email at kyle.johnson(at)givewp.com? I can send you a copy, but the repo/org doesn't have any available "seats" for an outside collaborator (or something).

**Comment by emanuellopess** [comment]
@kjohnson I sent an email on August 16, 2024, but I haven’t received a response. Did you receive my email? Do you have any feedback on this documentation?
